### PR TITLE
Model design: add all models without migration

### PR DESCRIPTION
**Pull request by jinnatul**
**What's New**
- Models
  - UserRole
  - Role
  - Student
  - StudentEnrollment
  - Submissions
  - User
  - Instructor
  - ClassRoom
  - ExpClass
  - Region
  - Country
  - Institution
  - InstLevelGroup
  - PreAuthList
  - Experiments
  - Subscription
  - Transaction
  - PaymentMethod
  - LevelGroup

### CODE REVIEW

File: api/Models/ApplicationDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        public DbSet<Subscription> Subscriptions { get; set; }
+
+        public DbSet<Transaction> Transactions { get; set; }
+
+        public DbSet<User> Users { get; set; }
+
+        public DbSet<UserRole> UserRoles { get; set; }
+
+    }
+}
```
Comment (shunjid) [inline api/Models/ApplicationDbContext.cs:52]: Need to override all the relations here.

File: api/Models/Role.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        [Key]
+        public string RoleId { get; set; }
+        
+        public string RoleName { get; set; }
+        
+        public string NormalizedName { get; set; }
+
+        public ICollection<UserRole> Roles { get; set; }
+    }
+}
```
Comment (shunjid) [inline api/Models/Role.cs:17]: Need to deal with IdentityRole. 
[Reference](https://stackoverflow.com/questions/22057403/extending-identityrole-and-identityuser)

File: api/Models/Student.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace api.Models
+{
+    public class Student
+    {
```
Comment (shunjid) [inline api/Models/Student.cs:7]: Where is the inheritance?

File: api/Models/User.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace api.Models
+{
+    public class User
+    {
+        [Key]
+        public string Id { get; set; }
+
```
Comment (shunjid) [inline api/Models/User.cs:10]: Need to associate **IdentityRole**

File: api/appsettings.json [hunk 5]
```diff
@@ -6,5 +6,8 @@
       "Microsoft.Hosting.Lifetime": "Information"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "ConnectionStrings": {
+    "ApplicationDbContext": "Data Source=eLuminousAPI.db"
+  }
```
Comment (shunjid) [inline api/appsettings.json:12]: Need to add Remote Database

**Review by shunjid — APPROVED** [review]
### Report
- Good effort
- Not satisfied due to relations and Identity (Role, User) not included
- Adding it to future-factory because needs strong review after review and security impelementation

**Review by zubayerhimel — APPROVED** [review]
Boss did all the required reviews, so I have nothing to say now. Good work 😄

**Comment by shunjid** [comment]
Boss left it incomplete @zubayerhimel xD

original

**Pull request merged**